sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/ListingFileCatalog.scala

```diff
@@ -50,6 +50,19 @@ class ListingFileCatalog(
 
   refresh()
 
+  /**
+   * Often HDFS create temporary files while copying to a new directory or writing new content.
+   * These files are unintentionally picked up by streaming - causing job failures. This option lets
+   * HDFS skip these files matching the configured regex-patterns from being picked up by Streaming
+   * Job.
+   */
+  private lazy val excludeFiles: Set[String] = parameters
+    .getOrElse("excludeFiles", ".*._COPYING_,_temporary").split(",").toSet
+
+  private def isExcludedFile(path: Path): Boolean = {
+    excludeFiles.map(path.getName.matches).fold(false)(_ || _)
```

Comment:
Probably better to precompile a regex here, as this method will be called a lot (100X per second X number of files under the root directory, by default)